third_party/gpus/cuda_configure.bzl

```diff
@@ -144,13 +144,20 @@ def _get_python_bin(repository_ctx):
 
 
 def _get_nvcc_tmp_dir_for_windows(repository_ctx):
-  """Return the tmp directory for nvcc to generate intermediate source files."""
+  """Return the Windows tmp directory for nvcc to generate intermediate source files."""
   escaped_tmp_dir = escape_string(
-      get_env_var(repository_ctx, "TMP", "C:\\Windows\\Temp").replace(
-          "\\", "\\\\"),)
+    get_env_var(repository_ctx, "TMP", "C:\\Windows\\Temp").replace(
+      "\\", "\\\\"),)
   return escaped_tmp_dir + "\\\\nvcc_inter_files_tmp_dir"
 
 
+def _get_nvcc_tmp_dir_for_unix(repository_ctx):
+  """Return the UNIX tmp directory for nvcc to generate intermediate source files."""
+  escaped_tmp_dir = escape_string(
+    get_env_var(repository_ctx, "TMPDIR", "/tmp"))
```

Comment:
Can you add `TMP` and `TMPDIR` here:
https://github.com/tensorflow/tensorflow/blob/c2e2ecedfef715b8f756c5e2df4dab3b88531253/third_party/gpus/cuda_configure.bzl#L1569-L1574

So that we the environment variable changes, the CROSSTOOL will get reconfigured.